### ISSUE TITLE
SVG Badge

### ISSUE THREAD

**Issue by lvillani** [gratipay/gratipay-badge#2]
Hi,

Would it be possible to have an SVG badge so that it renders well on HiDPI screens (as well as regular screens)?

**Comment by twolfson** [comment]
I am currently on break but if `<canvas>` supports exporting to SVG, then it should be possible. Care to explore it?

https://github.com/twolfson/gittip-badge/blob/0.2.1/lib/index.html#L33

**Comment by lvillani** [comment]
Looks like html2canvas has an SVG render but according to the comment at the beginning of the file (https://github.com/niklasvh/html2canvas/blob/master/src/renderers/SVG.js) it hasn't been tested for a while.

Before I delve in wrestling with html2canvas, is there a particular reason to generate the badge by going through PhantomJS instead of e.g.: working on an SVG within Inkscape and then exporting it?

Would you accept a pull request in which the source format becomes a plain SVG with tooling to export it as PNG?

**Comment by twolfson** [comment]
That is `html2canvas` function is for converting an `<svg>` in HTML into its `<canvas>` representation. The line I linked to is still the important one. We want to use `html` as the source since it allows us to use the GitHub CSS. The change should be as simple as

``` js
canvas.toDataUrl('image/svg');
// or whatever the alternative of `canvas.toDataUrl('image/png')` is for SVGs
```

We use PhantomJS because it allows for automation of the process, less tooling, allows for testing, and easier releases for developers. If we wind up generating an SVG, I would prefer to automate the conversion to a PNG (e.g. via `svg2png` or another tool) rather than making the developer take a manual step.

https://www.npmjs.org/package/svg2png

**Comment by lvillani** [comment]
I ended up creating a pure SVG badge with Inkscape in a repository of my own after struggling a bit to get an decent SVG out of canvas.

**Comment by twolfson** [comment]
Are you hosting a repo somewhere? If you are, I would love to add a link. If not, would you be open to contributing it to `gratipay-badge`? I am going to reopen this issue because this resolution doesn't scale for other developers.

**Comment by lvillani** [comment]
Yes but it still uses the old branding. I didn't even notice that Gittip changed name to Gratipay until I gave a look at github.com/dashboard/issues.

I will submit a pull request with the updated SVG (even though I'm planning to stop using Gittip/Gratipay).

**Comment by twolfson** [comment]
This has been resolved via #9 and released in `2.2.0`.
